Sources/SwiftSyntaxBuilder/DeclBuildables.swift.gyb

```diff
@@ -0,0 +1,83 @@
+%{
+  from gyb_syntax_support import *
+  from gyb_syntax_support.kinds import lowercase_first_word
+  from gyb_syntax_support.kinds import SYNTAX_BASE_KINDS
+  NODE_MAP = create_node_map()
+  # -*- mode: Swift -*-
+  # Ignore the following admonition it applies to the resulting .swift file only
+  
+  def syntax_buildable_child_type(type_name, syntax_kind, is_token, is_optional=False):
+    if syntax_kind in SYNTAX_BASE_KINDS:
+      buildable_type = syntax_kind + 'Buildable'
+    elif not is_token:
+      buildable_type = syntax_kind
+    else:
+      buildable_type = type_name
+
+    if is_optional:
+      buildable_type += '?'
+
+    return buildable_type
+}%
+// Automatically Generated From DeclBuildables.swift.gyb.
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+% for node in SYNTAX_NODES:
+%   if node.is_buildable():
+%     should_generate_init = False
+%     for child in node.children:
+%       child_node = NODE_MAP.get(child.syntax_kind)
+%       if child_node and child_node.is_syntax_collection():
+%         should_generate_init = True
+%       end
+%     end
+%     if should_generate_init:
+extension ${node.syntax_kind} {
+  init(
+%       init_parameters = []
+%       for child in node.children:
+%         child_node = NODE_MAP.get(child.syntax_kind)
+%         if child_node and child_node.is_syntax_collection():
+%           init_parameters.append("@%sBuilder %sBuilder: () -> %s = { .empty }" % (child.syntax_kind, child.swift_name, child.syntax_kind))
+%         else:
+%           param_type = syntax_buildable_child_type(child.type_name, child.syntax_kind, child.is_token(), child.is_optional)
+%           default_value = " = nil" if child.is_optional else ""
+%           init_parameters.append("%s: %s%s" % (child.swift_name, param_type, default_value))
+%         end
+%       end
+    ${',\n    '.join(init_parameters)}
+  ) {
+    self.init(
+%     init_parameters = []
+%     for child in node.children:
+%       child_node = NODE_MAP.get(child.syntax_kind)
+%       if child_node and child_node.is_syntax_collection():
+%         if child_node.collection_element is "Syntax":
```

Comment:
I think the `if` and `else` branch are doing the same thing, so you don’t need the condition.

Reply:
That was actually a question. 
We have one place [here](https://github.com/apple/swift-syntax/pull/284/files/4c2ea03308554b7c35071e78d4d0d2e62571da77#diff-30af98704bd9407c6b3ee049cc324ffe08d525bb80f1760107f81497aafafb20R84) etc where we have a TokenListBuilder. 
It returns a `TokenList`.
But it expects a `TokenListSynx` [here](https://github.com/apple/swift-syntax/blob/main/Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift#L1720)

So in this special case (I only found two places where it is used) we should maybe do something line 
`tokenListBuilder().buildTokenListSyntax(format: ...)`

So what do we need to do here? 
Should we ignore this case? 

It affects `ClosureCaptureItem` and `Attribute`

Reply:
`Cannot convert value of type 'TokenList' to expected argument type 'TokenListSyntax?'`

Reply:
I think your “here“ comments don’t contain line numbers. Could you add them to your comment. That way I don’t have to search 😉

Reply:
GitHub won't add line numbers for the first "here", so line 84 in DeclBuildables.swift 😑🙃
The `ClosureCaptureItem`

The second "here" should direct to the line

Reply:
I think [Buildables.swift:1720](https://github.com/apple/swift-syntax/blob/main/Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift#L1720) should take a `TokenList`, not a `TokenListSyntax`. Looking into this a bit I *believe* this is because `TokenListSyntax` is classified as a token in [Child.py:39](https://github.com/apple/swift/blob/main/utils/gyb_syntax_support/Child.py#L39), which is *probably* a bug and should be checking for a `Token` suffix instead of a `Token` substring. Could you maybe look into it?